pieman/bin/compress_image.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# Copyright (C) 2019 Denis Gavrilyuk <karpa4o4@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+"""
+
+import argparse
+import bz2
+import lzma
+import os
+
+
+COMPRESS_TYPES = {
+    'bzip2': (bz2.compress, 'bz2'),
+    'xz': (lzma.compress, 'xz')
+}
+
+
+def read_in_chunks(file_object, chunk_size=10485760):
+    """Lazy function (generator) to read a file piece by piece.
+    Default chunk size: 10M."""
+
+    while True:
+        data = file_object.read(chunk_size)
+        if not data:
+            break
+        yield data
+
+
+def main():
+    """The main entry point."""
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('compress_type', help='compress type {}'
+                        .format(tuple(COMPRESS_TYPES.keys())))
+    parser.add_argument('image', help='path to image')
+    args = parser.parse_args()
+
+    compress_type = args.compress_type
+    image = args.image
+
+    compressor, extension = COMPRESS_TYPES[compress_type]
+
+    with open(image, 'rb') as original:
+        generated_file = '{}.{}'.format(image, extension)
+
+        with open(generated_file, 'wb') as generated:
+            for compressing_chunk in read_in_chunks(original):
+                generated.write(compressor(compressing_chunk))
+
+            generated.close()
+        original.close()
```

Comment:
You don't have to close the files explicitly, because you're using the context managers (the `with` statement) here.